spotbugs/src/main/java/edu/umd/cs/findbugs/util/SecurityManagerHandler.java

```diff
@@ -0,0 +1,60 @@
+package edu.umd.cs.findbugs.util;
+
+/**
+ * Since Java 17, the security manager is deprecated for removal and invoking related methods
+ * causes a warning to be printed to the console. This intermediate disables the use of
+ * security manager-related APIs on Java 17 or later, unless using the security manager is
+ * explicitly configured by setting the <i>edu.umd.cs.findbugs.securityManagerDisabled</i>
+ * property.
+ */
+public class SecurityManagerHandler {
+
+    /**
+     * Determines if the security manager is used by SpotBugs.
+     */
+    public static boolean SECURITY_MANAGER_DISABLED;
+
+    static {
+        boolean securityManagerDisabled;
+        try {
+            String property = System.getProperty("edu.umd.cs.findbugs.securityManagerDisabled");
+            if (property != null) {
+                securityManagerDisabled = Boolean.parseBoolean(property);
+            } else {
+                String version = System.getProperty("java.version");
+                if (version.startsWith("1.")) {
+                    version = version.substring(2, 3);
+                } else {
+                    int index = version.indexOf(".");
+                    if (index != -1) {
+                        version = version.substring(0, index);
+                    }
+                }
+                securityManagerDisabled = Integer.parseInt(version) > 16;
+            }
+        } catch (Throwable ignored) {
```

Comment:
We're using Java 8 so let's consider to catch multiple concrete types instead of a single abstract `Throwable`:

```suggestion
        } catch (SecurityException | NumberFormatException ignored) {
```

Reply:
I don't quite agree as the check is supposed to handle any type of error to avoid backwards compatibility issues. With the Apache utilities, for example, a `NullPointerException` might be thrown without this being documented. If updating the code, one would need to adjust the catch block without this being obvious or compile-checked. A catch-all seems like the more appropriate option.